azurelinuxagent/common/protocol/metadata.py

```diff
@@ -170,7 +170,14 @@ def detect(self):
                                 "{0}.crt".format(thumbprint))
         shutil.copyfile(trans_prv_file, prv_file)
         shutil.copyfile(trans_cert_file, crt_file)
-        self.update_goal_state()
+        self._update_certs_with_retry()
+
+    def update_goal_state(self):
+        # Metadata doesn't cache the goal state; nothing to do here...
+        pass
+
+    def get_endpoint(self):
```

Comment:
The code was using a data member 'endpoint' on protocol objects, but the member did not exist in the Protocol interfaces. I added it as a getter.;